LabApiUtilities/src/main/com/microsoft/identity/labapi/utilities/authentication/KeyVaultCertificateBasedAuthenticationClient.java

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.labapi.utilities.authentication;
+
+import com.microsoft.identity.labapi.utilities.authentication.client.IConfidentialAuthClient;
+import com.microsoft.identity.labapi.utilities.authentication.common.CertificateCredential;
+import com.microsoft.identity.labapi.utilities.authentication.common.ClientCertificateMetadata;
+import com.microsoft.identity.labapi.utilities.authentication.common.KeyStoreConfiguration;
+import com.microsoft.identity.labapi.utilities.authentication.msal4j.Msal4jAuthClient;
+import com.microsoft.identity.labapi.utilities.exception.LabApiException;
+import com.microsoft.identity.labapi.utilities.exception.LabError;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import lombok.NonNull;
+
+/**
+ * A an authentication client that can acquire access tokens for Labs KeyVaults (rather than Labs API).
+ */
+public class KeyVaultCertificateBasedAuthenticationClient implements IAccessTokenSupplier {
+    private final static String KEYVAULT_SCOPE = "https://vault.azure.net/.default";
+    private final static String TENANT_ID = "72f988bf-86f1-41af-91ab-2d7cd011db47";
+    private final static String AUTHORITY = "https://login.microsoftonline.com/" + TENANT_ID;
+    private final static String CLIENT_ID = "f62c5ae3-bf3a-4af5-afa8-a68b800396e9";
+    private final static String KEYSTORE_TYPE = "Windows-MY";
+    private final static String KEYSTORE_PROVIDER = "SunMSCAPI";
+    private final static String CERTIFICATE_ALIAS = "LabVaultAccessCert";
```

Comment:
You can use the values in `LabConstants.java` for the cert alias and client id. Note that the alias used here now is not compatible with the new cert.

Reply:
Thanks. This callout probably saved me a lot of time :)